proto/viam/app/data/v1/data.proto

```diff
@@ -244,6 +247,19 @@ message DeleteTabularDataByFilterResponse {
   reserved "result";
 }
 
+// DeleteTabularDataRequest deletes the data from the organization that is older than `delete_before_days`.
+// For example if `delete_before_days` is 10, this deletes any data that was uploaded up to 10 days ago. If it is 0 or
+// unspecified, all existing data is deleted.
```

Comment:
I don't know if I love this default behavior. Mostly because of how easy it is to accidentally delete all your data by just omitting the delete_before_days field. Maybe users are already taking on that risk when hitting an API called "DeleteTabularData", idk. I just think I'd be more comfortable if we made this a pointer and then returned an error saying delete_before_days was a required field if it's null.

Reply:
I definitely get your point, but my only issue is whether it'd be strange from an API perspective to make this `optional` in order to make it a pointer, but then actually require it and error if not provided.

Reply:
I asked about this issue, and seems like there's not a great answer. You can read more [here](https://viaminc.slack.com/archives/C01Q5S8V6TB/p1693317871874619)! James suggested using oneof, so the user could provide a number > 0 or the string "delete_all" (or something like that). I don't feel too strongly so up to you to decide what's best, even if that's just leaving this as-is. Thanks for the discussion!

Reply:
Thanks for explaining! I have found oneof annoying to use in Go so I think that makes sense; I didn't realize we had the engineering guideline but glad we do. The example from MongoDB is helpful as another reference point too; this SGTM then.